pkg/cvo/updatepayload.go

```diff
@@ -189,13 +190,40 @@ func (r *payloadRetriever) fetchUpdatePayloadToDir(ctx context.Context, dir stri
 		},
 	}
 
+	// Prune older jobs while gracefully ignoring errors.
+	r.pruneJobs()
+
 	_, err := r.kubeClient.BatchV1().Jobs(job.Namespace).Create(job)
 	if err != nil {
 		return err
 	}
 	return resourcebuilder.WaitForJobCompletion(ctx, r.kubeClient.BatchV1(), job)
 }
 
+// pruneJobs deletes older jobs in the namespace while keeping the latest.
+func (r *payloadRetriever) pruneJobs() error {
+	jobs, err := r.kubeClient.BatchV1().Jobs(r.namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	if len(jobs.Items) <= 2 {
+		return nil
+	}
+
+	// Sort jobs by StartTime to determine the oldest
+	sort.Slice(jobs.Items, func(i, j int) bool {
+		return jobs.Items[i].Status.StartTime.Before(jobs.Items[j].Status.StartTime)
```

Comment:
Is start time guaranteed to be non empty and non nil?  Unstarted new jobs should be kept, not deleted

Reply:
Good catch. New jobs have a nil value immediately after they are created. Code is modified to no longer delete jobs that are new or still active.